### PR TITLE
 Move /examples and /dev to new mattermost-app-examples repository

### PR DESCRIPTION
**Pull request by azigler**
_New version of #357 based on #336_ 
 
**This PR should not be merged until https://github.com/mattermost/mattermost-developer-documentation/pull/1106 is merged.**

Fixes:
- https://github.com/mattermost/mattermost-plugin-apps/issues/330
- https://github.com/mattermost/mattermost-plugin-apps/issues/329
- https://github.com/mattermost/mattermost-plugin-apps/issues/300

Updates:
- `README` points to developer resources and guides

This PR moves:
- the content from `/examples` to the new https://github.com/mattermost/mattermost-app-examples repository.
- the `/dev` environment to the new https://github.com/mattermost/mattermost-app-examples repository
- the new content from merged PR https://github.com/mattermost/mattermost-plugin-apps/pull/352 to correspond with the new Python quick start guide in https://github.com/mattermost/mattermost-developer-documentation/pull/1106.

`/examples/goapp` has been preserved in this repository as `/goapp`.

### CODE REVIEW

**Comment by azigler** [comment]
@mickmister @levb Was simpler to make a new PR from latest master. Thanks y'all!

**Review by levb — CHANGES_REQUESTED** [review]
2 high-level comments:

1. The E2E (Cypress) tests depend on the hello-world app, so will be broken by this PR.

I suggest that we fix the E2E test (it's currently broken with the `master` webapp) and make it use the test app `/test/app` instead of hello.

2. As I understand, this PR creates a `/goapp` directory for the goapp examples. 3/5 that's either wrong, or we should move the `goapp` implementation there (currently in `/apps/goapp`) and have a `./examples` underneath.

File: README.md
Comment (mickmister) [inline README.md:15]:
```suggestion
Refer to the [Mattermost Apps Quick Start Guide](https://developers.mattermost.com/integrate/apps/quickstart/) for instructions on how to use the development environment and examples in the [mattermost/mattermost-app-examples](https://github.com/mattermost/mattermost-app-examples) repository.
```

**Comment by mickmister** [comment]
> 2. As I understand, this PR creates a `/goapp` directory for the goapp examples. 3/5 that's either wrong, or we should move the `goapp` implementation there (currently in `/apps/goapp`) and have a `./examples` underneath.

@levb When you say "this PR", do you mean this PR we're commenting on? The PR mostly removing things, and doesn't add/move any files. 1/5 I'm thinking we should have the `goapp` tooling in the Apps repo, and have an example implementation using it in the examples repo.

**Review by mickmister — COMMENTED** [review]
I see one typo in the copy changes, otherwise LGTM :+1:

Holding off on approving due to pending concerns from Lev

**Comment by levb** [comment]
@azigler fix lint, please?

**Comment by mickmister** [comment]
@levb This is good to merge now. Note that I temporarily put `examples/go/hello-world` in `test/hello-world` to make the e2e tests work. Do you think the test is still necessary?

**Comment by mickmister** [comment]
@levb I'd like you opinion on the above question before merging. cc @hanzei

**Comment by levb** [comment]
@mickmister 3/5 we should be using the test app for the E2E test and not hello. If you could make the change, I'd MUCH appreciate it.

**Comment by mickmister** [comment]
> @mickmister 3/5 we should be using the test app for the E2E test and not hello. If you could make the change, I'd MUCH appreciate it.

@levb Mind if I tackle this while taking care of the e2e webapp master branch issue? It seems out of scope of this PR

**Comment by azigler** [comment]
Thank you @mickmister and @levb!

**Comment by levb** [comment]
@mickmister @azigler then I'd argue don't bother with leaving hello-world since the test is broken ATM anyway; it doesn't even get to the point of deploying the app.

**Pull request merged**
